### PR TITLE
Updated file name for "High Availability of Deployed Components" topic

### PR DESCRIPTION
**Pull request by n-boshnakov**
**How to categorize this PR?**

/area documentation
/kind enhancement

**What this PR does / why we need it**:
This PR:
- updates file name for "High Availability of Deployed Components" topic
- fixes broken links due to name change

The name change is due to a collision with the `high-availability` folder name in the `usage` directory (https://github.com/gardener/gardener/tree/master/docs/usage/high-availability), leading to unstable behaviour in the public Gardener website.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```doc developer
NONE
```

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -157,7 +157,7 @@
 * [Extending the Monitoring Stack](development/monitoring-stack.md)
 * [How to create log parser for container into fluent-bit](development/log_parsers.md)
 * [`PriorityClasses` in Gardener Clusters](development/priority-classes.md)
-* [High Availability Of Deployed Components](development/high-availability.md)
+* [High Availability Of Deployed Components](development/high-availability-dev.md)
```
Comment (ialidzhikov) [inline docs/README.md]: Cosmetic suggestion:

```suggestion
* [High Availability Of Deployed Components](development/high-availability-of-components.md)
```

to make it closer to the document title.

Comment (n-boshnakov) [reply docs/README.md]: I agree with that suggestion. @Kostov6 What do you think?
EDIT: We are all in agreement to change the title.

**Review by LucaBernstein — COMMENTED** [review]
Thanks for the PR. There is another reference in https://github.com/gardener/gardener/blob/8ce7b065bfde5a1f943b71c504a36c5756667beb/pkg/gardenlet/operation/botanist/kubeapiserver.go#L81-L84

**Comment by n-boshnakov** [comment]
> Thanks for the PR. There is another reference in
> 
> https://github.com/gardener/gardener/blob/8ce7b065bfde5a1f943b71c504a36c5756667beb/pkg/gardenlet/operation/botanist/kubeapiserver.go#L81-L84

Thank you, I've resolved this.

**Review by LucaBernstein — COMMENTED** [review]
/lgtm

**Comment by ialidzhikov** [comment]
```
[FAILED] Timed out after 10.001s.
  Expected success, but got an error:
      <*errors.StatusError | 0xc000560280>: 
      servicemonitors.monitoring.coreos.com "garden-gardener-operator" not found
      {
          ErrStatus: {
              TypeMeta: {Kind: "", APIVersion: ""},
              ListMeta: {
                  SelfLink: "",
                  ResourceVersion: "",
                  Continue: "",
                  RemainingItemCount: nil,
              },
              Status: "Failure",
              Message: "servicemonitors.monitoring.coreos.com \"garden-gardener-operator\" not found",
              Reason: "NotFound",
              Details: {
                  Name: "garden-gardener-operator",
                  Group: "monitoring.coreos.com",
                  Kind: "servicemonitors",
                  UID: "",
                  Causes: nil,
                  RetryAfterSeconds: 0,
              },
              Code: 404,
          },
      }
  In [It] at: /home/prow/go/src/github.com/gardener/gardener/test/integration/operator/garden/garden/garden_test.go:[437](https://prow.gardener.cloud/view/gs/gardener-prow/pr-logs/pull/gardener_gardener/10658/pull-gardener-integration/1846530038447476736#1:build-log.txt%3A437) @ 10/16/24 12:37:05.743
```

Occurrence of https://github.com/gardener/gardener/issues/10576

---

/test pull-gardener-integration

**Comment by ialidzhikov** [comment]
```
  [FAILED] Expected
      <string>: v1.106.0-dev
  to equal
      <string>: v1.105.0
  In [It] at: /home/prow/go/src/github.com/gardener/gardener/test/e2e/gardener/upgrade/upgrade.go:232 @ 10/16/24 13:01:45.401
```

---

/test pull-gardener-e2e-kind-upgrade

**Comment by ialidzhikov** [comment]
```
[FAILED] Timed out after 10.000s.
  Expected success, but got an error:
      <*errors.StatusError | 0xc00093d540>: 
      servicemonitors.monitoring.coreos.com "garden-gardener-operator" not found
      {
          ErrStatus: {
              TypeMeta: {Kind: "", APIVersion: ""},
              ListMeta: {
                  SelfLink: "",
                  ResourceVersion: "",
                  Continue: "",
                  RemainingItemCount: nil,
              },
              Status: "Failure",
              Message: "servicemonitors.monitoring.coreos.com \"garden-gardener-operator\" not found",
              Reason: "NotFound",
              Details: {
                  Name: "garden-gardener-operator",
                  Group: "monitoring.coreos.com",
                  Kind: "servicemonitors",
                  UID: "",
                  Causes: nil,
                  RetryAfterSeconds: 0,
              },
              Code: 404,
          },
      }
  In [It] at: /home/prow/go/src/github.com/gardener/gardener/test/integration/operator/garden/garden/garden_test.go:[437](https://prow.gardener.cloud/view/gs/gardener-prow/pr-logs/pull/gardener_gardener/10658/pull-gardener-integration/1846550354112548864#1:build-log.txt%3A437) @ 10/16/24 13:58:02.622
```

---

/test pull-gardener-integration

**Pull request merged**